# Description of analyses

## Variant calling in ancient herring samples

Variant sites were identified separately for ancient and modern samples, using the *mpileup* and *call* (--consensus-caller model) 
commands in *bcftools version 1.9* (http://samtools.github.io/bcftools/call-m.pdf). Ancient genotypes were filtered with vcftools (Danecek et al. 2011) 
for genotype quality (--minQ 900) and SNPs were removed from the data set if they were genotyped in fewer than 75% of ancient samples. 

### Index the herring genome using samtools
FYI, this index is different from that used by *bowtie2* in earlier steps

``` bash
GENOMEDIR=/media/ubuntu/Herring_aDNA/atlantic_herring_genome # Path to directory with genome.
REF=GCA_900700415.1_Ch_v2.0.2_genomic.fna # Genome fasta file name
#####

cd $GENOMEDIR
samtools faidx $REF
```
### Using *bcftools version 1.9*, use *mpileup* command to create multi-way pileup and calculate genotype likelihoods.

explanation of terms:

mpileup:      multi-way pileup producing genotype likelihoods

--fasta-ref FILE    faidx indexed reference sequence file

--bam-list FILE     list of input BAM filenames, one per line

--skip-indels       do not perform indel calling

--output FILE       write output to FILE [standard output]

--output-type TYPE  'b' compressed BCF; 'u' uncompressed BCF; 'z' compressed VCF; 'v' uncompressed VCF [v]

--annotate: depth-related vcf tags are generated by mpileup, see bcftools mpileup --annotate ? for list.

FORMAT/AD  .. Allelic depth (Number=R,Type=Integer)

FORMAT/DP  .. Number of high-quality bases (Number=1,Type=Integer)

INFO/AD  .. Total allelic depth (Number=R,Type=Integer)


``` bash
BAMDIR=/media/ubuntu/hybridization_capture/ancient_samples/mapdamage_bam #directory with sorted, indexed, filtered, and base-recalibrated .bam files
GENOMEDIR=/media/ubuntu/Herring_aDNA/atlantic_herring_genome # Path to directory with genome.
REF=GCA_900700415.1_Ch_v2.0.2_genomic.fna # Genome fasta file name
BAMLIST=mybamlist.txt
OUTFILE=mpileup_results.bcf
#####

#1. Create list of bam files and save to a txt file:

cd $BAMDIR
ls *.bam>$BAMLIST

#2. Create multi-way pileup 

bcftools mpileup --fasta-ref $GENOMEDIR'/'$REF \
--bam-list $BAMDIR'/'$BAMLIST \
--skip-indels \
--output $BAMDIR'/'$OUTFILE \
--output-type b \
--annotate FORMAT/AD,FORMAT/DP,INFO/AD

```

### Using *bcftools version 1.9*, use *call* command to do SNP variant calling from VCF/BCF file created in the previous step (mpileup)

Usage:   bcftools call [options] <in.vcf.gz>

--variants-only            output variant sites only

--consensus-caller       command to call genotypes

--output-type <b|u|z|v>     output type: 'b' compressed BCF; 'u' uncompressed BCF; 'z' compressed VCF; 'v' uncompressed VCF [v]

--output <file>             write output to a file [standard output]
  
-A, --keep-alts              keep all possible alternate alleles at variant sites

-f, --format-fields <list>      output format fields: GQ,GP

``` bash
BAMDIR=/media/ubuntu/hybridization_capture/ancient_samples/mapdamage_bam #directory with sorted, indexed, filtered, and base-recalibrated .bam files
INFILE=mpileup_results.bcf
OUTFILE=ancient_call_results.bcf
####

cd $BAMDIR
bcftools call --variants-only \
--consensus-caller \
--keep-alts \
--format-fields GQ \
--output-type b \
--output $OUTFILE \
$INFILE

```
