# Description of analyses

## Variant calling in ancient herring samples

Variant sites were identified separately for ancient and modern samples, using the *mpileup* and *call* (--consensus-caller model) 
commands in *bcftools version 1.9* (http://samtools.github.io/bcftools/call-m.pdf). Ancient genotypes were filtered with vcftools (Danecek et al. 2011) 
for genotype quality (--minQ 900) and SNPs were removed from the data set if they were genotyped in fewer than 75% of ancient samples. 

### Index the herring genome using samtools
FYI, this index is different from that used by *bowtie2* in earlier steps

``` bash
GENOMEDIR=/media/ubuntu/Herring_aDNA/atlantic_herring_genome # Path to directory with genome.
REF=GCA_900700415.1_Ch_v2.0.2_genomic.fna # Genome fasta file name
#####

cd $GENOMEDIR
samtools faidx $REF
```
### Using *bcftools version 1.9*, use *mpileup* command to create multi-way pileup and calculate genotype likelihoods.

explanation of terms:

mpileup:      multi-way pileup producing genotype likelihoods

--fasta-ref FILE    faidx indexed reference sequence file

--bam-list FILE     list of input BAM filenames, one per line

--skip-indels       do not perform indel calling

--output FILE       write output to FILE [standard output]

--output-type TYPE  'b' compressed BCF; 'u' uncompressed BCF; 'z' compressed VCF; 'v' uncompressed VCF [v]

--annotate: depth-related vcf tags are generated by mpileup, see bcftools mpileup --annotate ? for list.

FORMAT/AD  .. Allelic depth (Number=R,Type=Integer)

FORMAT/DP  .. Number of high-quality bases (Number=1,Type=Integer)

INFO/AD  .. Total allelic depth (Number=R,Type=Integer)


``` bash
BAMDIR=/media/ubuntu/hybridization_capture/ancient_samples/mapdamage_bam #directory with sorted, indexed, filtered, and base-recalibrated .bam files
GENOMEDIR=/media/ubuntu/Herring_aDNA/atlantic_herring_genome # Path to directory with genome.
REF=GCA_900700415.1_Ch_v2.0.2_genomic.fna # Genome fasta file name
#####

#2. Create list of bam files and save to a txt file:
ls *.bam>mybamlist.txt


bcftools mpileup --fasta-ref GCA_900700415.1_Ch_v2.0.2_genomic.fna --bam-list mybamlist.txt --skip-indels --output mpileup_results.bcf --output-type b --annotate FORMAT/AD,FORMAT/DP,INFO/AD

```


